collect_app/src/main/java/org/odk/collect/android/database/DatabaseConnection.kt

```diff
@@ -26,29 +27,28 @@ open class DatabaseConnection(
     val readableDatabase: SQLiteDatabase
         get() = dbHelper.readableDatabase
 
-    private val dbHelper: SQLiteOpenHelper by lazy {
-        getOpenHelper(path + name) {
-            DatabaseMigratorSQLiteOpenHelper(
-                AltDatabasePathContext(path, context),
-                name,
-                null,
-                databaseVersion,
-                migrator
-            )
+    private val dbHelper: SQLiteOpenHelper
+        get() {
+            val databasePath = path + File.separator + name
+            if (openHelpers.containsKey(databasePath) && !File(databasePath).exists()) {
+                openHelpers.remove(databasePath)
+            }
+
+            return openHelpers.getOrPut(databasePath) {
```

Comment:
Maybe we could just use `if-else` (either get one of put and return) here instead of `getOrPut` now? I think that'd be clearer.

Reply:
`getOrPut` is better because it's null safe with if-else the syntax would look worse unless I'm missing something.

Reply:
Ah you're right we'd always end up having to deal with the nullable result of `get`